### DISCUSSION TITLE
meta.isBlurred Becomes true on Value Change Before Field is Actually Blurred

### DISCUSSION THREAD

**Discussion by lroskoshin** [TanStack/form#1034] [General]
Hey everyone,

I’ve noticed an issue when using onChange validation in TanStack Form.

When I use onChange validation, the `meta.isBlurred` property behaves unexpectedly. Specifically, if I just change the value of a field, isBlurred becomes true even before I actually blur the field.

This seems like a bug because it prevents me from implementing good UX. I want to show errors on blur but hide them when the user starts to change the field.
	•	If I use `onChange` validation, the field displays an error as soon as the user types the first letter.
	•	If I use `onBlur` validation, the field remains invalid even if the user fixes the mistake but hasn’t blurred the field yet.

I believe this issue might be related to the code at this line:
[FormApi.ts Line 626](https://github.com/TanStack/form/blob/1db18b2b72ff64338fe614c3558ba722b09223bc/packages/form-core/src/FormApi.ts#L626)

Could someone please look into this?

Thank you!

**Comment by Pascalmh** [comment]
See https://github.com/TanStack/form/pull/940
